packages/at_client/lib/src/client/at_client_impl.dart

```diff
@@ -517,8 +520,15 @@ class AtClientImpl implements AtClient, AtSignChangeListener {
     }
     // Transform put request
     // Optionally passing encryption private key to sign the public data.
-    UpdateVerbBuilder verbBuilder = await PutRequestTransformer(this)
-        .transform(tuple, encryptionPrivateKey: encryptionPrivateKey);
+    UpdateVerbBuilder verbBuilder = await putRequestTransformer.transform(tuple,
+        encryptionPrivateKey: encryptionPrivateKey);
+    // Validate the size of the value after encryption/encoding
+    // Since AtClientPreference is mandatory argument in create method, _preference
+    // will not be null.
+    if (verbBuilder.value.length > _preference!.maxDataSize) {
```

Comment:
similar check can be done in remoteSecondary.executeCommand. if final command to secondary greater than what secondary can accept

Reply:
Added check in executeCommand in RemoteSecondary.